src/resources/js/admin-log-controls.js

```diff
@@ -0,0 +1,171 @@
+var tribe_logger_admin = tribe_logger_admin || {};
+var tribe_logger_data  = tribe_logger_data || {};
+
+( function( $, obj ) {
+	var working        = false;
+	var current_view   = '';
+	var current_engine = '';
+	var view_changed   = false;
+	var $controls      = $( '#tribe-log-controls' );
+	var $options       = $controls.find( 'select' );
+	var $spinner       = $controls.find( '.working' );
+	var $viewer        = $( '#tribe-log-viewer' );
+	var $download_link = $( 'a.download_log' );
+
+	/**
+	 * Update the log view based on changes to the various selectors.
+	 */
+	function update() {
+		// If an update is already in progress let's wait until that job completes
+		if ( working ) {
+			return;
+		}
+
+		detect_view_change();
+		freeze();
+		request();
+	}
+
+	/**
+	 * Communicate any changes back to the server so we can obtain fresh log data
+	 * to display to the user.
+	 */
+	function request() {
+		var data = {
+			'action':     'tribe_logging_controls',
+			'check':      tribe_logger_data.check,
+			'log-level':  $( '#log-level' ).find( ':selected' ).attr( 'name' ),
+			'log-engine': $( '#log-engine' ).find( ':selected' ).attr( 'name' )
+		};
+
+		if ( view_changed ) {
+			data['log-view'] = current_view;
+		}
+
+		$.ajax( ajaxurl, {
+			'method':   'POST',
+			'success':  on_success,
+			'error':    on_error,
+			'dataType': 'json',
+			'data':     data
+		} );
+	}
+
+	/**
+	 * Unfreeze the controls (following an update) and refresh on-screen data as needed.
+	 *
+	 * @param data
+	 */
+	function on_success( data ) {
+		unfreeze();
+
+		if ( $.isArray( data.data.entries ) ) {
+			$viewer.html( to_table( data.data.entries ) );
+			update_download_link();
+		}
+	}
+
+	/**
+	 * Converts data_array, which is expected to be an array of
+	 * arrays, into an HTML table.
+	 */
+	function to_table( data_array ) {
+		var html = '<table>';
+
+		for ( var row in data_array ) {
+			html += '<tr>';
+
+			for ( var cell in data_array[row] ) {
+				html += '<td>' + data_array[row][cell] + '</td>';
+			}
+
+			html += '</tr>';
+		}
+
+		return html + '</table>';
+	}
+
+	/**
+	 * Add, append or update the log download link so it points to the
+	 * currently selected log file.
+	 */
+	function update_download_link() {
+		var url = $download_link.attr( 'href' );
+		var log = encodeURI( get_current_view() );
+		var matches = url.match(/&log=([a-z0-9\-]+)/i);
+
+		// Update or add the log parameter
```

Comment:
Like this idea though I do, my inclination at this stage in the game is to defer it and do it as a follow-on piece of work (for timeliness, but also because it impacts more than just the logging framework).